roles/backup/tasks/secrets.yml

```diff
@@ -0,0 +1,124 @@
+---
+
+- name: Make _secrets directory
+  file:
+    path: "_secrets"
+    state: directory
+    mode: '0700'
+
+- name: Get secret_key
+  k8s_info:
+    kind: Secret
+    namespace: '{{ meta.namespace }}'
+    name: '{{ tower_secret_key_secret }}'
+  register: _secret_key
+
+- name: Set secret key
+  set_fact:
+    secret_key: "{{ _secret_key['resources'][0]['data']['secret_key'] | b64decode }}"
+
+- name: Template secret_key definition
+  template:
+    src: secret_key_secret.yml.j2
+    dest: "_secrets/secret_key_secret.yml"
+    mode: '0600'
+
+- name: Set secret key template
+  set_fact:
+    secret_key_template: "{{ lookup('file', '_secrets/secret_key_secret.yml') }}"
+
+- name: Write secret_key to pvc
+  community.kubernetes.k8s_exec:
+    namespace: "{{ meta.namespace }}"
+    pod: "{{ meta.name }}-db-management"
+    command: >-
+      bash -c "echo '{{ secret_key_template }}' > {{ _backup_dir }}/secret_key_secret.yml"
+
+- name: Get admin_password
+  k8s_info:
+    kind: Secret
+    namespace: '{{ meta.namespace }}'
+    name: '{{ tower_admin_password_secret }}'
+  register: _admin_password
```

Comment:
I think this will only work in the case where the admin password was generated for the user as its using the default which assumes naming conventions: https://github.com/ansible/awx-operator/pull/133/files#diff-f9dcb95da85ffdf5274f2bf0f6f1dfea4073036a82af584a51b4e981af2a0e60R15

Instead you could gather the admin password from the status object of the AWX CR. This would ensure you are getting the value from either the supplied or generated admin password secret: https://github.com/ansible/awx-operator/blob/devel/ansible/templates/crd.yml.j2#L270

Reply:
I take it back I guess you allow the override via CR input here: https://github.com/ansible/awx-operator/pull/133/files#diff-ea6ce699660ccb70f61f22b9377eeec75ac9b63126ec970e7c9e90d441be415bR44-R46

My only question is wouldn't it be cleaner if you had these secrets (https://github.com/ansible/awx-operator/pull/133/files#diff-ea6ce699660ccb70f61f22b9377eeec75ac9b63126ec970e7c9e90d441be415bR41-R52) updated in the status of the AWX CR and then just needed the CR name to be able to pull them out and back them up?

Reply:
Thanks for the suggestion @chambridge, I used your idea of updating the AWX CR with the secret names as statuses.  I added that and use those status names during the backup role now: 
* [Set Status](https://github.com/ansible/awx-operator/pull/133/files#diff-da6ce4fb98bdf57987f09e9b524e00f6600a03d966e5121c7c87c103351abec4R43-R50)
* [Use status to retrieve secret values](https://github.com/ansible/awx-operator/pull/133/files#diff-a29bce4ae2beeb7039f9e579e788bf0cdcf975514ce5e7674f5e2b3ba0984e58R3-R8)